grammar.js

```diff
@@ -1,58 +1,149 @@
 module.exports = grammar({
   name: 'help', // The actual language name is help
 
-  extras: ($) => [/[\t ]/],
-
-  externals: ($) => [
-    $.word,
-    $._imm_word,
-    $.column_name,
-    $.uppercase_name,
-    $.code_block
-  ],
+  extras: () => [/[\t ]/],
 
   rules: {
     help_file: ($) =>
-      repeat1(choice($.line, '\n', $.column_heading, $.headline, $.code_block)),
+      seq(
+        repeat(/[\t ]*\n/),  // Eat whitespace at top of file.
+        repeat(choice(
+          $.block,
+          $.column_heading,
+          $.headline,
+        )),
+        // Last block may not be followed by a blank line or EOL.
+        optional($.block_end), // alias($.block_end, $.block)
+      ),
 
     headline: ($) =>
       seq(
-        field('delimiter', choice(/===.*===/, /---.*---/)),
-        '\n',
+        field('delimiter', choice(
+          seq(/[\t ]*============+[\t ]*\n/)),
+          seq(/[\t ]*------------+[\t ]*\n/)),
         repeat1($._atom),
         '\n',
+        optional(_blank()),
       ),
 
-    line: ($) => prec.right(seq(repeat1($._atom), optional($.code_block), optional('\n'))),
+    // Text block/paragraph: adjacent lines followed by blank line(s).
+    block: ($) =>
+      seq(
+        repeat1(choice($.line, $.line_li)),
+        _blank(),
+      ),
+    // Special case: last block in the document may not end with blank line (nor even EOL).
+    block_end: ($) => choice(
+      choice($.line_noeol, $.line_li_noeol),
+      seq(repeat1(choice($.line, $.line_li)), choice($.line_noeol, $.line_li_noeol))),
+
+    // Code block: adjacent lines surrounded by ">" and "<".
+    // TODO: expose (word)s in code (line)s? But whitespace must be preserved.
+    code_block: ($) => seq(
+      />[\t ]*\n/,
+      repeat(alias($.line_code, $.line1)),
+      alias($.line_code_noeol, $.line2),
```

Comment:
Can't those two be renamed to `line` ?

Reply:
yes, will alias these to "line" now that `code_block` is working.